vterm.el

```diff
@@ -831,29 +836,67 @@ the called functions."
         (apply (cadr f) args)
       (message "Failed to find command: %s" command))))
 
+
+(defun vterm--prompt-tracking-enabled-p ()
+  "Check whether feature of tracking the prompt is enabled or not.
+
+Prompt tracking need shell side configurations.
+
+For zsh user, this is done by PROMPT=$PROMPT'%{$(vterm_prompt_end)%}'.
+
+The shell send semantic information about where the prompt ends via properly
+escaped sequences to Emacs.
+
+More information see `Shell-side configuration' and `Directory tracking'
+in README."
+  (or vterm--prompt-tracking-enabled-p
+      (save-excursion
+        (setq vterm--prompt-tracking-enabled-p
+              (next-single-property-change (point-min) 'vterm-prompt)))))
+
+(defun vterm-next-prompt (n)
+  "Move to end of Nth next prompt in the buffer."
+  (interactive "p")
+  (if (vterm--prompt-tracking-enabled-p)
+      (let ((pt (point)) )
+        (goto-char (vterm--get-prompt-point))
+        (cl-loop repeat (or n 1) do
+                 (setq pt (next-single-property-change (point-at-bol 2) 'vterm-prompt))
+                 (when pt (goto-char pt))))
+    (term-next-prompt n)))
+
+(defun vterm-previous-prompt (n)
+  "Move to end of Nth previous prompt in the buffer."
+  (interactive "p")
+  (if (vterm--prompt-tracking-enabled-p)
+      (let ((pt (point)) )
+        (goto-char (vterm--get-prompt-point))
+        (cl-loop repeat (or n 1) do
+                 (setq pt (previous-single-property-change (1- (point)) 'vterm-prompt))
+                 (when pt (goto-char (1- pt)))))
+    (term-previous-prompt n)))
+
 (defun vterm--get-prompt-point ()
-  "Get the position of the end of current prompt."
+  "Get the position of the end of current prompt.
+More information see `vterm--prompt-tracking-enabled-p' and
+`Directory tracking and Prompt tracking'in README. "
   (let (pt)
     (save-excursion
-      (setq pt (vterm--get-prompt-point-internal
-                vterm--term (line-number-at-pos))))
+      (end-of-line)
+      (if (get-text-property (point) 'vterm-prompt)
+          (setq pt (point))
+        (setq pt (previous-single-property-change (point) 'vterm-prompt))
```

Comment:
This `previous-single-property-change` return nil every time :( 

Either vterm-prompt is not getting set in the c code, or libvterm is not actually tracking.